### PR TITLE
feat(plugin-users-permissions): auth email case-insensitive

### PR DESCRIPTION
**Pull request by lucaperret**

### CODE REVIEW

File: packages/strapi-plugin-users-permissions/controllers/Auth.js [hunk 1]
```diff
@@ -32,10 +32,11 @@ module.exports = {
       const isEmail = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(params.identifier);
 
       // Set the identifier to the appropriate query field.
+      const identifier = params.identifier.toLowerCase();
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-users-permissions/controllers/Auth.js]: Why do you need to lowercase the identifier at this line? It should only concern the email. Also, if we do that we have to be sure that the stored email is lowercased during the registration process or that our query whatever the connector (strapi-bookshelf, strapi-mongoose) are making a query that is case-insensitive.

**Comment by Aurelsicoko** [comment]
Thank you for your work @lucaperret! It's great to see active contributor like you 👍 

Can you read my comment and come back to us with the necessary updates?

File: packages/strapi-plugin-users-permissions/controllers/Auth.js (hunk 1 above)
Comment (lucaperret) [reply packages/strapi-plugin-users-permissions/controllers/Auth.js]: The identifier is lowercased here:
https://github.com/strapi/strapi/pull/442/commits/d9198dd225dcd4f37c8615031e48ce19208d6e36
Do you prefer to only lowercase the email?

Comment (Aurelsicoko) [reply packages/strapi-plugin-users-permissions/controllers/Auth.js]: Yes, it's a mistake on our side. We should also remove the spaces, accents, etc by using the [_.deburr()](https://lodash.com/docs/4.17.4#deburr) method of Lodash. But, we have to keep uppercase characters.

Comment (lucaperret) [reply packages/strapi-plugin-users-permissions/controllers/Auth.js]: ok so I'll apply lowercase only in email field.

Comment (Aurelsicoko) [reply packages/strapi-plugin-users-permissions/controllers/Auth.js]: @lucaperret Awesome! We will be able to merge after these changes.

Comment (lucaperret) [reply packages/strapi-plugin-users-permissions/controllers/Auth.js]: Ok but how to lowercase only the email field ?
Here https://github.com/strapi/strapi/pull/442/commits/d9198dd225dcd4f37c8615031e48ce19208d6e36 I don't know if it's an email or username...

Comment (Aurelsicoko) [reply packages/strapi-plugin-users-permissions/controllers/Auth.js]: @lucaperret You have to add this line to the register method (https://github.com/strapi/strapi/blob/d9198dd225dcd4f37c8615031e48ce19208d6e36/packages/strapi-plugin-users-permissions/controllers/Auth.js#L32)

**Comment by lucaperret** [comment]
Done

**Pull request merged**